ldms/src/sampler/dstat/dstat.c

```diff
@@ -173,7 +173,11 @@ static char * compute_pidopts_schema(struct attr_value_list *kwl, struct attr_va
 	if (dosuffix) {
 		snprintf(hbuf, bsz, "_%x", pidopts);
 	}
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpragmas"
+#pragma GCC diagnostic ignored "-Wformat-truncation"
```

Comment:
I think this is a real bug? bsz is 21B. hbuf and buf are both 21B. If you concatenate those two you get 42B which results in the truncation warning.

Reply:
Thanks for catching this. Let me put in a proper fix.